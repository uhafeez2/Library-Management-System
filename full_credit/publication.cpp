 #include "publication.h"
 #include <string>
 #include <iostream>

 using namespace std;


 bool Publication::is_checked_out() {return checked_out;}
 void Publication::check_out(Patron& p_patron) {
   if (checked_out) {
     throw Invalid_transaction();
   } else {
     patron = p_patron;
     checked_out = true;
   }
 }
   
 void Publication::check_in() {
   if (checked_out) {
     checked_out = false;
   } else {
     throw Invalid_transaction();
   }
 }

 string Publication::to_string() {
   string pub = "\"" + title + "\"" + " by " + author + ", " + copyright + 
     " (" + target_age.to_string() + " " + genre.to_string() + " "  
         + media.to_string() + ") " + "ISBN: " + isbn;
   if (checked_out) {
      pub += "\nChecked out to " + patron.get_patron_name() + 
             " (" + patron.get_patron_phone_number() + ")";
   }
   return pub;
 }
