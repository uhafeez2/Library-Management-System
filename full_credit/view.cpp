#include "view.h"
#include "library.h"
#include "controller.h"
#include <iostream>
#include <string>
#include <vector>
#include <exception>
#include <FL/Fl.H>
#include <FL/Fl_Window.H>
#include <FL/Fl_Shared_Image.H>
#include <FL/fl_ask.H>

using namespace std;

string View::menu() {
  string menu = R"(
=================================
CSE1325 Library Management System
=================================

Publications
------------
(1) Add publication
(2) List all publications
(3) Check out publication
(4) Check in publication

Patrons
=======
(5) Add patron
(6) List all patrons

Utility
-------
(9) Help
(0) Exit

)";
	return menu;
}

string View::list_publication() {
  string list = R"(
----------------------------
List of Library Publications
----------------------------
)";
  //cout << header;
  for (int i=0; i<library.number_of_publications(); ++i) {
    list+=std:: to_string(i) + ") " + library.publication_to_string(i) + '\n';
  }
return list;
}

string View::list_patrons() {
  string list = R"(
-----------------------
List of Beloved Patrons
-----------------------
)";
  //cout << header;
  for (int i=0; i<library.number_of_patrons(); ++i) {
     list+=std::to_string(i) + ") " + library.patron_to_string(i) + '\n';
  }
return list;
}

string View::list_genre(){
string list = "";
  for (int i = 0; i < Genre::num_genres; ++i)
    list += "  " + std::to_string(i) + ") " + Genre(i).to_string()+'\n';
return list;
}

string View::list_media() {
  string list = "";
  for (int i = 0; i < Media::num_media; ++i)
    list += "  " + std::to_string(i) + ") " + Media(i).to_string() +'\n';
 return list;
}

string View::list_age() {
  string list = "";
  for (int i = 0; i < Age::num_ages; ++i)
   list += "  " + std::to_string(i) + ") " + Age(i).to_string() + '\n';
return list;
}



string View::get_help() {
  return ( "The LMS tracks publication assets for a library, including those who check out and return those publications. \n(1) Add publication - This allows the librarian to enter data associated with a new publication. \n(2) List all publications - All data known about each publication in the library is listed.\n(3) Check out publication - Enter the data for patrons who check out a publication, and mark that publication as checked out.\n(4) Check in publication - Select a publication and mark it as checked in.\n(5) Add patron - This allows the librarian to enter data associated with a new library patron.\n(6) List patrons - All data know about each patron of the library. \n (9) Help - Print this text.\n(0) Exit - Exit the program. WARNING: The current version does NOT save any entered data. This feature will be added in the next version." );
  
}


