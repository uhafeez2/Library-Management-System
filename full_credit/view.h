#ifndef __VIEW_H
#define __VIEW_H 201609
 
#include "library.h"

class View {
  public:
    View(Library& lib) : library(lib) { }
    string menu();//void show_menu();
    string list_publication();//void list_publications();
    string list_patrons();//void list_patrons();
    string list_age();
    string list_genre();
    string list_media();
    string get_help(); //void help();
  private:
    Library& library; 
};
#endif
