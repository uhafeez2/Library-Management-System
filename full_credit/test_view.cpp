#include "view.h"
#include "patron.h"
#include "publication.h"
#include "library.h"
#include <iostream>
using namespace std;

int main() {
  Library library;
  View view(library);

  bool passed = true;

  //
  // Set up library
  //
  Patron larry("Larry", "817-555-1111");
  Patron curly("Curly", "817-555-2222");
  Patron moe("Moe", "817-555-3333");

  library.add_patron(larry);
  library.add_patron(curly);
  library.add_patron(moe);

  Publication foundation("Foundation", "Isaac Asimov", "1942", 
      Genre::fiction, Media::book, Age::adult, "0385177259");
  Publication foundation_empire("Foundation and Empire", "Isaac Asimov", "1943", 
      Genre::fiction, Media::book, Age::adult, "0385177259");
  Publication foundation2("Second Foundation", "Isaac Asimov", "1944", 
      Genre::fiction, Media::book, Age::adult, "0385177259");
  Publication war("War of the Worlds", "Jeff Wayne", "1977",
      Genre::performance, Media::audio, Age::teen, "9780711969148");
  Publication wonka("Willy Wonka and the Chocolate Factory", "Roald Dahl", "1971",
      Genre::performance, Media::video, Age::children, "0142410314");

  library.add_publication(foundation);
  library.add_publication(foundation_empire);
  library.add_publication(foundation2);
  library.add_publication(war);
  library.add_publication(wonka);

  library.check_out(2, 1);

  //
  // Test menu
  //
  cout << endl;
  view.show_menu();

  //
  // Test list publications
  //
  cout << endl;
  view.list_publications();

  //
  // Test list patrons
  //
  cout << endl;
  view.list_patrons();

  //
  // Show results
  //
  cout << endl << "Verify output for pass or fail" << endl;
}
