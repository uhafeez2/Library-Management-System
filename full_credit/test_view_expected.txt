

=================================
CSE1325 Library Management System
=================================

Publications
------------
(1) Add publication
(2) List all publications
(3) Check out publication
(4) Check in publication

Patrons
=======
(5) Add patron
(6) List all patrons

Utility
-------
(9) Help
(0) Exit



----------------------------
List of Library Publications
----------------------------
0) "Foundation" by Isaac Asimov, 1942 (adult fiction book) ISBN: 0385177259
1) "Foundation and Empire" by Isaac Asimov, 1943 (adult fiction book) ISBN: 0385177259
2) "Second Foundation" by Isaac Asimov, 1944 (adult fiction book) ISBN: 0385177259
Checked out to Curly (817-555-2222)
3) "War of the Worlds" by Jeff Wayne, 1977 (teen performance audio) ISBN: 9780711969148
4) "Willy Wonka and the Chocolate Factory" by Roald Dahl, 1971 (children performance video) ISBN: 0142410314


-----------------------
List of Beloved Patrons
-----------------------
0) Larry (817-555-1111)
1) Curly (817-555-2222)
2) Moe (817-555-3333)

Verify output for pass or fail
